p2p/client.go

```diff
@@ -313,6 +337,36 @@ func (c *Client) setupPeerDiscovery(ctx context.Context) error {
 	return nil
 }
 
+func (c *Client) setupBlockedPeers(peers []peer.AddrInfo) error {
+	if c.gater == nil {
+		return fmt.Errorf("connection gater is nil, cannot blacklist peers")
```

Comment:
As there are no arguments here, please use `errors.New` instead of `fmt.Errorf`.